src/components/MAPI/clusters/utils.ts

```diff
@@ -843,3 +851,63 @@ export function getClusterUpdateSchedule(
     targetTime: parseRFC822DateFormat(targetTime),
   };
 }
+
+export enum ClusterStatus {
+  Idle = 'IDLE',
+  DeletionInProgress = 'DELETION_IN_PROGRESS',
+  CreationInProgress = 'CREATION_IN_PROGRESS',
+  UpgradeInProgress = 'UPGRADE_IN_PROGRESS',
+  UpgradeScheduled = 'UPGRADE_SCHEDULED',
+  UpgradeAvailable = 'UPGRADE_AVAILABLE',
+}
+
+export function useClusterStatus(
+  cluster?: capiv1beta1.ICluster,
+  providerCluster?: ProviderCluster,
+  releases?: releasev1alpha1.IRelease[]
+): { status?: ClusterStatus; clusterUpdateSchedule?: IClusterUpdateSchedule } {
```

Comment:
Could we maybe put the hook in its own file? That way it's easier to distinguish between it and the other utility functions. Just a suggestion 🙂